sdk/azcore/response.go

```diff
@@ -129,6 +129,16 @@ func (r *Response) UnmarshalAsXML(v interface{}) error {
 	return err
 }
 
+// UnmarshalError will unmarshal the received service error payload along with the corresponding unmarshal error.
+// NOTE: this method is only meant to be called when handling a service error.
+func (r *Response) UnmarshalError(unmarshalErr error) error {
+	payload, err := r.payload()
```

Comment:
I believe it's possible (although unlikely) that `payload` could be empty.  If this is the case, we have two options.

- add some default message "missing body" or whatever
- just return `unmarshalErr`

I _think_ that the JSON/XML unmarshallers will fail with a message indicating the first case ("unexpected end of blah blah").  If that's the case then let's do the second one.

Reply:
I will check this and simply return the unmarshalErr if the other unmarshallers have a good message.

Reply:
It doesnt seem that the other unmarshallers add a message if the payload is empty here's one of them as reference:
```
func (r *Response) UnmarshalAsJSON(v interface{}) error {
	payload, err := r.payload()
	if err != nil {
		return err
	}
	// TODO: verify early exit is correct
	if len(payload) == 0 {
		return nil
	}
	err = r.removeBOM()
	if err != nil {
		return err
	}
	err = json.Unmarshal(payload, v)
	if err != nil {
		err = fmt.Errorf("unmarshalling type %T: %s", v, err)
	}
	return err
}
```

Reply:
I think it's still a good idea to just include the unmarshalErr even if the payload is empty, since the err we're sending is populated with the error from json.Ummarshal()

Reply:
I added empty error body to the message. Done?

Reply:
Oh yes I forgot we have that empty body check.  We should follow up on that separately to see if that's really what we should be doing (I forget why I added that, might have been early on prototyping).